### ISSUE TITLE
GRASP: Genome-Wide Repository of Associations Between SNPs and Phenotypes

### ISSUE THREAD

**Issue by Shicheng-Guo** [ramiromagno/gwasrapidd#30]
Hi Ramiro, 

Any plan to add GRASP to the package? 

https://grasp.nhlbi.nih.gov/Overview.aspx

Thanks. 

Shicheng

**Comment by Shicheng-Guo** [comment]
another interesting dataset will be: https://genebass.org/

**Comment by ramiromagno** [comment]
Hi Shicheng,

Thank you for your question and interest in `{gwasrapidd}`.

Those two sources of GWAS you mentioned are interesting. However, I do not have plans to provide access to them through gwasrapidd. 

Generally speaking, you may access these types of sources in three distinct ways:

1. Access through a web app. This type of data access is more akin to end user access, i.e., of a more manual type of browsing and exploring of the data.
2. Access by a REST API service, which allows programmatic analysis of the data and is usually programming language agnostic and more used by bioinformaticians.
3. Access by downloading an entire dump of the database, which also allows programmatic analysis but requires you to store all data in your own disk, and usually requires more effort to understand the data schema (also ideal for bioinformaticians).

The GWAS Catalog provides all three approaches, and `{gwasrapidd}` facilitates the access via the REST API method. Now, these two other sources you mentioned do not seem to provide access via a REST API. And even if they did, the way data is organized is likely different from the way the data is organized in GWAS Catalog. So, for these reasons, I do not think that it would make sense to try to aggregate such heterogeneous functionality under one single package.

In summary, I think it would be helpful to have the data from GRASP and genebass easily accessible into R, but that effort probably requires two dedicated R packages. :) If more people would show interest, and if a couple of programming hands would be available to work on such a challenge, I would not mind joining efforts.

**Comment by Shicheng-Guo** [comment]
Agree with you Ramiro!! I am sure there will be some similar R packages coming out for GRASP and Genebass. Will notify you when I find them. Thank you for these impressive comments!!

Shicheng

**Comment by ramiromagno** [comment]
You're welcome. The answer to your other question follows next. :)
